src/derivative_operators/concretization.jl

```diff
@@ -247,3 +247,91 @@ LinearAlgebra.Array(Q::ComposedMultiDimBC, Ns) = Tuple(Array.(Q.BCs, Ns))
 SparseArrays.SparseMatrixCSC(Q::ComposedMultiDimBC, Ns...) = Tuple(sparse.(Q.BCs, Ns))
 SparseArrays.sparse(Q::ComposedMultiDimBC, Ns) = SparseMatrixCSC(Q, Ns)
 BandedMatrices.BandedMatrix(Q::ComposedMultiDimBC, Ns) = Tuple(BandedMatrix.(Q.BCs, Ns))
+
+# HIgher Dimensional Concretizations. The following concretizations return two dimensional arrays
+# which operate on flattened vectors. Mshape is the size of the unflattened array on which A is operating on.
+
+function LinearAlgebra.Array(A::DerivativeOperator{T,N}, Mshape) where {T,N}
+    # Case where A is not differentiating along the first dimension
+    if N != 1
+        n = 1
+        for M_i in Mshape[1:N-1]
+            n *= M_i
+        end
+        B = Kron(Array(A), Eye(n))
+        if N != length(Mshape)
+            n = 1
+            for M_i in Mshape[N+1:end]
+                n *= M_i
+            end
+            B = Kron(Eye(n), B)
+        end
+
+    # Case where A is differentiating along hte first dimension
+    else
+        n = 1
+        for M_i in Mshape[2:end]
+            n *= M_i
+        end
+        B = Kron(Eye(n), Array(A))
+    end
+    return BandedBlockBandedMatrix(B)
+end
+
+function SparseArrays.SparseMatrixCSC(A::DerivativeOperator{T,N}, Mshape) where {T,N}
+    # Case where A is not differentiating along the first dimension
+    if N != 1
+        n = 1
+        for M_i in Mshape[1:N-1]
+            n *= M_i
+        end
+        B = Kron(sparse(A), sparse(I,n,n))
+        if N != length(Mshape)
+            n = 1
+            for M_i in Mshape[N+1:end]
+                n *= M_i
+            end
+            B = Kron(sparse(I,n,n), B)
+        end
+
+    # Case where A is differentiating along hte first dimension
+    else
+        n = 1
+        for M_i in Mshape[2:end]
+            n *= M_i
+        end
+        B = Kron(sparse(I,n,n), sparse(A))
+    end
+    return BandedBlockBandedMatrix(B)

```

Comment:
Should return a SparseMatrixCSC

Reply:
Can it still be a lazy array? As in a Kron product of SparseMatrixCSC's or must it completely be a SparseMatrixCSC?

Reply:
We want to return the array type the user asks for if they use the constructor. This is because they make assumptions that only hold for specific array types (which is why they directly called the constructor). The form that we should be smart with is the `sparse` function.